src/reconcile/rec_write.c

```diff
@@ -1463,40 +1479,20 @@ __rec_txn_read(WT_SESSION_IMPL *session, WT_RECONCILE *r,
 #else
 	timestampp = NULL;
 #endif
-	all_visible = *updp == first_txn_upd && !uncommitted &&
+	all_visible = upd == first_txn_upd && !uncommitted &&
```

Comment:
This is a little subtle -- `*updp` may have diverged from `upd` at this point.  I'm open to adding a boolean local variable at [line 1427](https://github.com/wiredtiger/wiredtiger/pull/3850/files#diff-a6c8d99189e5865e35b893d6792f1165R1427) if you think that would be clearer.